### PR TITLE
Other types

### PR DESCRIPTION
**Pull request by locks**
Replaces https://github.com/tildeio/simple-html-tokenizer/pull/35.
Ignored build stuff, but uses typing work.

cc @chadhietala @rwjblue

### CODE REVIEW

File: src/evented-tokenizer.ts [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
+    this.entityParser = entityParser;
+
+    this.state = null;
+    this.input = null;
+
+    this.index = -1;
+    this.line = -1;
+    this.column = -1;
+    this.tagLine = -1;
+    this.tagColumn = -1;
```
Comment (locks) [inline src/evented-tokenizer.ts]: should reset be called here?

File: src/entity-parser.ts [hunk 2]
```diff
@@ -3,7 +3,11 @@ const CHARCODE    = /^#([0-9]+)$/;
 const NAMED       = /^([A-Za-z0-9]+)$/;
 
 export default class EntityParser {
-  constructor(private named) {}
+  private named: {};
+
+  constructor(named) {
```
Comment (rwjblue) [inline src/entity-parser.ts]: If you leave it the way it was, this is the default.

So basically:

```ts
class Foo {
  constructor(private named) {}
}
```

And this:

```ts
class Foo {
  private named: {};

  constructor(named) {
    this.named = named;
  }
}
```

Are the same, but the first version is much shorter and has less repetition...

See [output in the playground](http://www.typescriptlang.org/play/index.html#src=class%20Foo%20%7B%0D%0A%20%20constructor(private%20named)%20%7B%7D%0D%0A%7D).

File: src/evented-tokenizer.ts [hunk 3]
```diff
@@ -1,23 +1,32 @@
...
+  private state: any;
+  private input: any;
+  private index: number;
+  private tagLine: number;
+  private tagColumn: number;
+
+  public line: number;
+  public column: number;
+
+  constructor(delegate, entityParser) {
```
Comment (rwjblue) [inline src/evented-tokenizer.ts]: `constructor(private: delegate, private: entityParser) {}` would be all that is required if the other private state is initialized where it is declared

File: src/evented-tokenizer.ts [hunk 4]
```diff
@@ -1,23 +1,32 @@
...
-  this.column = -1;
-  this.tagLine = -1;
-  this.tagColumn = -1;
-
-  this.reset();
-}
+export default class EventedTokenizer {
+  private delegate: any;
+  private entityParser: any;
+  private state: any;
```
Comment (rwjblue) [inline src/evented-tokenizer.ts]: I _think_ you can set the default value also: `private state: any = null;` (same for the other cases that are being initialized in the constructor now).

File: src/tokenize.ts [hunk 5]
```diff
@@ -1,8 +1,8 @@
-import Tokenizer from './tokenizer';
+import { default as Tokenizer, Token } from './tokenizer';
```
Comment (rwjblue) [inline src/tokenize.ts]: This should be:

```js
import Tokenizer, { Token } from './tokenizer';
```

File: src/tokenizer.ts [hunk 6]
```diff
@@ -1,39 +1,71 @@
...
+      column: number;
+    }
+  };
 }
 
-Tokenizer.prototype = {
-  tokenize: function(input) {
+export default class Tokenizer {
+  private token: Token;
+  private startLine: number;
```
Comment (rwjblue) [inline src/tokenizer.ts]: initialize here

File: src/tokenizer.ts [hunk 7]
```diff
@@ -1,39 +1,71 @@
...
+  private startColumn: number;
+  private options: TokenizerOptions;
+  private tokenizer: EventedTokenizer;
+  private tokens: Token[] = [];
+  private currentAttribute: Attribute = null;
+
+  constructor(entityParser, options: TokenizerOptions = {}) {
```
Comment (rwjblue) [inline src/tokenizer.ts]: Should be able to do:

```
constructor(entityParser, private options: TokenizerOptions = {}) {
  this.tokenizer = new EventedTokenizer(this, entityParser);
}
```

File: src/utils.ts [hunk 8]
```diff
@@ -2,14 +2,14 @@ const WSP = /[\t\n\f ]/;
 const ALPHA = /[A-Za-z]/;
 const CRLF = /\r\n?/g;
 
-export function isSpace(char: string): boolean {
```
Comment (rwjblue) [inline src/utils.ts]: Why remove these types?

File: src/utils.ts [hunk 9]
```diff
@@ -2,14 +2,14 @@ const WSP = /[\t\n\f ]/;
...
+export function isSpace(char) {
   return WSP.test(char);
 }
 
-export function isAlpha(char: string): boolean {
```
Comment (rwjblue) [inline src/utils.ts]: ?

File: src/utils.ts [hunk 10]
```diff
@@ -2,14 +2,14 @@ const WSP = /[\t\n\f ]/;
...
+export function isAlpha(char) {
   return ALPHA.test(char);
 }
 
-export function preprocessInput(input: string): string {
```
Comment (rwjblue) [inline src/utils.ts]: ?

File: tests/tokenizer-tests.js [hunk 11]
```diff
@@ -1,4 +1,5 @@
-/* global HTML5Tokenizer: false */
+/*globals HTML5Tokenizer */
```
Comment (rwjblue) [inline tests/tokenizer-tests.js]: can you put a space after `/*`?

**Pull request merged**